### PR TITLE
feat(questionnaire-web-app): submit reply to horizon

### PR DESCRIPTION
**Pull request by swdpcomputing**
## Ticket Number

https://pins-ds.atlassian.net/browse/AS-

## Checklist

- [ ] Requires infrastructure changes
- [ ] If adding or remove environment variables (e.g. in `docker-compose.yaml`) then I have updated the appropriate Helm chart
- [ ] I have updated the documentation accordingly
- [x] My commit history in this PR is linear
- [x] New features have tests
- [ ] Breaking change (team conversation required)

## Important

Please do not merge from `master` (please only [rebase](https://github.com/foundry4/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

### CODE REVIEW

File: packages/horizon-householder-appeal-publish/src/publishDocuments.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+ * Publish Documents
+ *
+ * Publishes the documents to Horizon. This is done asynchronously
+ * as we're not interested in the response
+ *
+ * @param log
+ * @param {{id: string, type: string}[]} documents
+ * @param {string} serviceId
+ * @param {string} horizonId
+ * @returns {Promise<void>}
```
Comment (a6software) [inline packages/horizon-householder-appeal-publish/src/publishDocuments.js:14]: I'm not sure this signature is correct.

The function doesn't return anything, so potentially just `void` ? 

Maybe wrong on this.

File: packages/appeal-reply-service-api/package.json [hunk 2]
```diff
@@ -22,18 +22,18 @@
...
     "compression": "^1.7.4",
-    "date-fns": "^2.16.1",
     "express": "^4.17.1",
     "express-async-errors": "^3.1.1",
     "express-pino-logger": "^5.0.0",
     "js-yaml": "^3.14.0",
     "mongodb": "^3.6.3",
     "mongoose": "^5.11.15",
     "neat-csv": "^5.2.0",
+    "notifications-node-client": "^5.1.0",
```
[the remark this answers is not in the document]
Comment (davehlewis) [reply packages/appeal-reply-service-api/package.json:32]: this was added due to throwing constant errors due to dependency issues from common module

File: packages/horizon-householder-appeal-publish/handler-reply.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+// ***** MAIN FUNCTION ***** //
+
+/**
+ * Handler Reply
+ * Publishes the documents to Horizon
+ *
+ * @param {object} event contains body.reply
+ * @param {string} horizonCaseId
+ * @param {string} context contains the httpStatusCode
+ * @returns {string} horizonCaseId or error message
```
[the remark this answers is not in the document]
Comment (swdpcomputing) [reply packages/horizon-householder-appeal-publish/handler-reply.js:70]: I like it. edited.

File: packages/horizon-householder-appeal-publish/src/callHorizon.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+ *
+ * Calls Horizon and gets the ID of the created
+ * case
+ *
+ * @param log
+ * @param input
+ * @returns {Promise<string>}
+ */
+const callHorizon = async (log, input) => {
+  log.info(input, 'Calling Horizon');
```
[the remark this answers is not in the document]
Comment (swdpcomputing) [reply packages/horizon-householder-appeal-publish/src/callHorizon.js:15]: This is actually Simon's code. I just split this out form one function for readability. I don't fancy changing too much as horizon code breaks easily.

File: packages/horizon-householder-appeal-publish/src/callHorizon.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+  const { data } = await axios.post('/horizon', input, {
+    baseURL: config.horizon.url,
+  });
+
+  log.info({ data }, 'Horizon response');
```
[the remark this answers is not in the document]
Comment (swdpcomputing) [reply packages/horizon-householder-appeal-publish/src/callHorizon.js:21]: This is actually Simon's code. I just split this out form one function for readability. I don't fancy changing too much as horizon code breaks easily.

File: packages/horizon-householder-appeal-publish/src/getLpaData.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+const config = require('./config');
+
+/**
+ * Get LPA Data
+ *
+ * Returns the LPA data
+ *
+ * @param log
+ * @param code
+ * @returns {Promise<{ id: string, name: string, inTrial: boolean, england: boolean, wales: boolean, horizonId: string }>}
```
[the remark this answers is not in the document]
Comment (swdpcomputing) [reply packages/horizon-householder-appeal-publish/src/getLpaData.js:11]: As above.

File: packages/horizon-householder-appeal-publish/src/getLpaData.js [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+ */
+
+const getLpaData = async (log, code) => {
+  log.info({ code }, 'Getting LPA data from Appeals Service API');
+
+  const { data } = await axios.get(`/api/v1/local-planning-authorities/${code}`, {
+    baseURL: config.appealsService.url,
+  });
```
[the remark this answers is not in the document]
Comment (swdpcomputing) [reply packages/horizon-householder-appeal-publish/src/getLpaData.js:19]: As above.

File: packages/horizon-householder-appeal-publish/src/publishDocuments.js (hunk 1 above)
Comment (swdpcomputing) [reply packages/horizon-householder-appeal-publish/src/publishDocuments.js:14]: I'm with you. Removed entirely.

**Review by Russ-Clarke — DISMISSED** [review]

**Review by Russ-Clarke — APPROVED** [review]

**Pull request merged**